src/multi.c

```diff
@@ -102,7 +102,10 @@ bool MultiHandleCommand(RedisRaftCtx *rr,
          */
         unsigned int cmd_flags = CommandSpecTableGetAggregateFlags(rr->commands_spec_table, cmds, 0);
 
-        if (cmd_flags & CMD_SPEC_UNSUPPORTED) {
+        if (validateCommandACL(ctx, cmds->commands[0]) == REDISMODULE_ERR) {
```

Comment:
Why are we manually validating here? Dry-run will validate commands in the array anyway, no?

Reply:
I believe (might be wrong) that multi evaluates at "append" to queue time as well, but perhaps I confused myself on that.  will check again

Reply:
chekecd the code, and it validates acl twice, at enqueue time and at "execution" time (in case ACL changed per comment in code)

Reply:
You're right and I'm not happy with that :laughing:

It is just weird. I wonder if there is any use-case for this. User won't see the ACL error until EXEC in any case. So, the only difference between "ACL check twice" and   "ACL check on EXEC" is when there is an ACL change while accumulating.

This scenario doesn't make any sense to me. Maybe I'm missing something but I personally don't think it is worth the overhead. In RR, now we have extra code for it (`validateCommandACL()`) and that function also adds performance overhead unnecessarily (due to allocations, lookup etc. Redis doesn't have any of these overheads). 

Let's keep it this way for now but discuss it tomorrow. It'd be great if we can get away without doing it twice.